site/docs/guide/context.md

```diff
@@ -169,62 +169,163 @@ The same context object for one update will be shared by all installed middlewar
 > If you are new to context objects, you don't need to worry about the rest of this page.
 
 You can install your own properties on the context object if you want.
-This is possible in two ways:
 
-1. Installing [middleware](./middleware.md) that modifies the context (recommended), or
-2. setting a custom context constructor.
+### Via Middleware (Recommended)
 
-If you choose option 1., you must specify the custom context as a type parameter (skip for JavaScript):
+The customizations can easily be done in [middleware](./middleware.md).
 
-<CodeGroup>
-  <CodeGroupItem title="Node.js" active>
+::: tip Middlewhat?
+This section requires an understanding of middleware, so in case you did not skip ahead to [this section](./middleware.md) yet, here is a very brief summary.
```

Comment:
```suggestion
This section requires an understanding of middleware, so in case you have not skipped ahead to [this section](./middleware.md) yet, here is a very brief summary.
```